kani-driver/src/call_cargo.rs

```diff
@@ -32,9 +32,14 @@ impl KaniSession {
             .target_dir
             .as_ref()
             .unwrap_or(&metadata.target_directory.clone().into())
-            .clone();
+            .clone()
+            .join("kani");
         let outdir = target_dir.join(build_target).join("debug/deps");
 
+        // Clean directory before building since we are unable to handle cache today.
+        // TODO: https://github.com/model-checking/kani/issues/1736
+        self.cargo_clean(&target_dir)?;
```

Comment:
Would deleting the `<target>/kani` directory instead work?

Reply:
Sure. `cargo clean` seems like a good way to clean stuff generated by `cargo build`, but the effect is the same, so if you prefer, I can just remove the folder.

Reply:
Thanks. I think the advantage of just removing the kani directory is to not wipe out the artifacts of a normal (non-kani) `cargo build`.

Reply:
Which would hopefully make the UX a bit "less-terrible".

Reply:
BTW, we were providing the correct target directory to `cargo clean`, so it was only wiping out `target/kani` not the entire `target`. Anyway, not a biggie.

Reply:
Sorry, I missed that.